completely refactor README & parts of docs (#3097)
docker-mailserver/docker-mailserver: docs/content/examples/tutorials/basic-installation.md

title: 'Tutorials | Basic Installation'
---

## A Basic Example With Relevant Environmental Variables

This example provides you only with a basic example of what a minimal setup could look like. We **strongly recommend** that you go through the configuration file yourself and adjust everything to your needs. The default [docker-compose.yml](https://github.com/docker-mailserver/docker-mailserver/blob/master/docker-compose.yml) can be used for the purpose out-of-the-box, see the [_Usage_ chapter](../../usage.md).

``` YAML

> version: '3.8'
> 
> services:

Reviewer: No version needed.

```suggestion
services:
```

  mailserver:
    image: docker.io/mailserver/docker-mailserver:latest
    container_name: mailserver
    # Provide the FQDN of your mail server here
    hostname: mail.example.com
    ports:
      - "25:25"
      - "143:143"
      - "587:587"
      - "993:993"
    volumes:
      - ./docker-data/dms/mail-data/:/var/mail/
      - ./docker-data/dms/mail-state/:/var/mail-state/
      - ./docker-data/dms/mail-logs/:/var/log/mail/
      - ./docker-data/dms/config/:/tmp/docker-mailserver/
      - /etc/localtime:/etc/localtime:ro
    environment:
      - ENABLE_SPAMASSASSIN=1
      - SPAMASSASSIN_SPAM_TO_INBOX=1
      - ENABLE_CLAMAV=1
      - ENABLE_FAIL2BAN=1
      - ENABLE_POSTGREY=1
      - ENABLE_SASLAUTHD=0
      - ONE_DIR=1
    cap_add:
      - NET_ADMIN
    restart: always
```

## A Basic LDAP Setup

**Note** There are currently no LDAP maintainers. If you encounter issues, please raise them in the issue tracker, but be aware that the core maintainers team will most likely not be able to help you. **We would appreciate and we encourage everyone to actively participate in maintaining LDAP-related code by becoming a maintainer!**

``` YAML

> version: '3.8'
> 
> services:

Reviewer: ```suggestion
services:
```

  mailserver:
    image: docker.io/mailserver/docker-mailserver:latest
    container_name: mailserver

>     # Provide the FQDN of your mail server her

Reviewer: ```suggestion
    # Provide the FQDN of your mail server here (Your DNS MX record should point to this value)
```

    hostname: mail.example.com
    ports:
      - "25:25"

>       - "143:143"

Reviewer: ```suggestion
```

      - LDAP_QUERY_FILTER_DOMAIN=(|(&(mail=*@%s)(objectClass=PostfixBookMailAccount)(mailEnabled=TRUE))(&(mailGroupMember=*@%s)(objectClass=PostfixBookMailAccount)(mailEnabled=TRUE))(&(mailalias=*@%s)(objectClass=PostfixBookMailForward)))
      - DOVECOT_PASS_FILTER=(&(objectClass=PostfixBookMailAccount)(uniqueIdentifier=%n))
      - DOVECOT_USER_FILTER=(&(objectClass=PostfixBookMailAccount)(uniqueIdentifier=%n))
      - ENABLE_SASLAUTHD=1
      - SASLAUTHD_MECHANISMS=ldap
      - SASLAUTHD_LDAP_SERVER=ldap
      - SASLAUTHD_LDAP_BIND_DN=cn=admin,dc=localhost,dc=localdomain
      - SASLAUTHD_LDAP_PASSWORD=admin
      - SASLAUTHD_LDAP_SEARCH_BASE=ou=people,dc=localhost,dc=localdomain
      - SASLAUTHD_LDAP_FILTER=(&(objectClass=PostfixBookMailAccount)(uniqueIdentifier=%U))
      - POSTMASTER_ADDRESS=postmaster@localhost.localdomain
      - POSTFIX_MESSAGE_SIZE_LIMIT=100000000
    cap_add:
      - NET_ADMIN
    restart: always
```

## A Detailed Example

!!! note

    This is a community contributed guide. Please let us know via a Github Issue if you're having any difficulty following the guide so that we can update it.

>     This section probably needs an overhaul!

Reviewer 2: laughed on that 😄 sure you want to keep it?

Reviewer: Removed overhaul line. Guide is fine for now, could do with a bit of revision but good enough.

Author: Haha, this was supposed to be a comment for maintainers/me, but I see what you mean. @polarathene is correct in removing it :D

> 
>     This section probably needs an overhaul!

Reviewer: ```suggestion
```

          mailserver:
            image: docker.io/mailserver/docker-mailserver:latest
            container_name: mailserver
            # provide the FQDN of your mail server here
            hostname: mail.example.com
            ports:
              - "25:25"
              - "143:143"
              - "587:587"
              - "993:993"
            volumes:
              - ./docker-data/dms/mail-data/:/var/mail/
              - ./docker-data/dms/mail-state/:/var/mail-state/
              - ./docker-data/dms/mail-logs/:/var/log/mail/
              - ./docker-data/dms/config/:/tmp/docker-mailserver/
              - /etc/localtime:/etc/localtime:ro
            environment:

>               - ENABLE_SPAMASSASSIN=1
>               - SPAMASSASSIN_SPAM_TO_INBOX=1
>               - ENABLE_CLAMAV=1

Reviewer: ```suggestion
```

              - ENABLE_FAIL2BAN=1

>               - ENABLE_POSTGREY=1
>               - ENABLE_SASLAUTHD=0

Reviewer: Bringing back original ENV:

```suggestion
              # Using letsencrypt for SSL/TLS certificates
              - SSL_TYPE=letsencrypt
              # Allow sending emails from other docker containers
              # Beware creating an Open Relay: https://docker-mailserver.github.io/docker-mailserver/edge/config/environment/#permit_docker
              - PERMIT_DOCKER=network
              # You may want to enable this: https://docker-mailserver.github.io/docker-mailserver/edge/config/environment/#spoof_protection
              # See step 8 below, which demonstrates setup with enabled/disabled SPOOF_PROTECTION:
              - SPOOF_PROTECTION=0
```

              - ONE_DIR=1
            cap_add:

>               - NET_ADMIN

Reviewer: ```suggestion
              - NET_ADMIN # For Fail2Ban to work
```

>             restart: always

Reviewer: Resolved: Reverted config
